core/templates/pages/topic-editor-page/modal-templates/rearrange-skills-in-subtopics-modal.component.spec.ts

```diff
@@ -0,0 +1,275 @@
+    expect(component.oldSubtopicId).toEqual(1);
+  });
+
+  it('should not call TopicUpdateService when skill is moved to same subtopic',
+    () => {
+      const dragAndDropEventClass = (
+        new DragAndDropEventClass<ShortSkillSummary>());
+      const containerData = [
+      {} as ShortSkillSummary
+      ];
+      const dragDropEvent = dragAndDropEventClass.createInContainerEvent(
+        'selectedItems', containerData, 1, 0);
+      let removeSkillSpy = spyOn(topicUpdateService, 'removeSkillFromSubtopic');
+      component.onMoveSkillEnd(dragDropEvent, null);
+      expect(removeSkillSpy).not.toHaveBeenCalled();
+    });
+
+  it('should call TopicUpdateService when skill is moved', () => {
+    const event = {
+      previousIndex: 1,
+      currentIndex: 2,
+      previousContainer: {
+        data: ['1', '2']
+      },
+      container: {
+        data: ['3']
+      },
+      item: {}
+    } as unknown as CdkDragDrop<ShortSkillSummary[]>;
+    let moveSkillSpy = spyOn(topicUpdateService, 'moveSkillToSubtopic');
+    component.onMoveSkillEnd(event, 1);
+    expect(moveSkillSpy).toHaveBeenCalled();
+  });
+
+  it('should call TopicUpdateService when skill is removed from subtopic',
+    () => {
+      const event = {
+        previousIndex: 1,
+        currentIndex: 1,
+        previousContainer: {
+          data: ['1', '2']
+        },
+        container: {
+          data: ['1']
+        },
+        item: {}
+      } as unknown as CdkDragDrop<ShortSkillSummary[]>;
+      let removeSkillSpy = spyOn(topicUpdateService, 'removeSkillFromSubtopic');
+      component.oldSubtopicId = 1;
+      component.onMoveSkillEnd(event, null);
+      expect(removeSkillSpy).toHaveBeenCalled();
+    });
+
+  it('should call TopicUpdateService when new Subtopic Id is null',
+    () => {
+      const event = {
+        previousIndex: 1,
+        currentIndex: 1,
+        previousContainer: {
+          data: ['1']
+        },
+        container: {
+          data: ['1']
+        },
+        item: {}
+      } as unknown as CdkDragDrop<ShortSkillSummary[]>;
+
+      let removeSkillSpy = spyOn(topicUpdateService, 'removeSkillFromSubtopic');
+      component.oldSubtopicId = null;
+      component.onMoveSkillEnd(event, null);
+      expect(removeSkillSpy).not.toHaveBeenCalled();
+    });
+
+  it('should not call TopicUpdateService if subtopic name validation fails',
+    () => {
+      component.editableName = 'subtopic1';
+      let subtopicTitleSpy = spyOn(topicUpdateService, 'setSubtopicTitle');
+      component.updateSubtopicTitle(1);
+      expect(subtopicTitleSpy).not.toHaveBeenCalled();
+    });
+
+  it('should call TopicUpdateService to update subtopic title', () => {
+    let subtopicTitleSpy = spyOn(topicUpdateService, 'setSubtopicTitle');
+    component.updateSubtopicTitle(1);
+    expect(subtopicTitleSpy).toHaveBeenCalled();
+  });
+
+  it('should call set and reset the selected subtopic index', () => {
```

Comment:
Please split the tests into (setup, pre-check, action, post-check) by adding a new line between them. Thanks! Please do this for all the tests